### PR TITLE
XW-2026 Create wds-dropdown using contextual components

### PR DESCRIPTION
**Pull request by rogatty**
## Links

* https://wikia-inc.atlassian.net/browse/XW-2026
* http://emberjs.com/blog/2016/01/15/ember-2-3-released.html#toc_contextual-components
* http://emberjs.com/api/classes/Ember.Templates.helpers.html#method_component

## Description

Use an idea from [ember-basic-dropdown](https://github.com/cibernox/ember-basic-dropdown/blob/master/addon/templates/components/basic-dropdown.hbs) to create `wds-dropdown` component which is reusable and potentially highly customizable.

The simplest way to use it is:
```
{{#design-system/wds-dropdown as |dropdown|}}
	{{#dropdown.toggle}}Toggle{{/dropdown.toggle}}
	{{#dropdown.content}}Content{{/dropdown.content}}
{{/design-system/wds-dropdown}}
```

## Reviewers

@Wikia/x-wing

### CODE REVIEW

**Comment by Grunny** [comment]
LGTM.

File: front/main/app/templates/components/design-system/global-navigation-dropdown.hbs [hunk 1]
```diff
@@ -1,21 +1,28 @@
...
-				{{else}}
-					{{design-system/global-navigation-link-dropdown model=link}}
-				{{/if}}
-			</li>
-		{{/each}}
-	</ul>
-</div>
+{{#design-system/wds-dropdown additionalClassNames=additionalClassNames as |dropdown|}}
+	{{#dropdown.toggle
+		additionalClassNames='wds-global-navigation__dropdown-toggle'
```
Comment (kvas-damian) [inline front/main/app/templates/components/design-system/global-navigation-dropdown.hbs]: can we just use `class` attribute here?

Comment (kvas-damian) [inline front/main/app/templates/components/design-system/global-navigation-dropdown.hbs]: I see we use `additionalClassNames` in many places and I believe we can simplify the code by using `class` attribute which is supported by Ember by default.

File: front/main/app/templates/components/design-system/global-navigation-account-navigation.hbs [hunk 2]
```diff
@@ -1,12 +1,16 @@
...
+{{#design-system/wds-dropdown additionalClassNames=additionalClassNames as |dropdown|}}
+	{{#dropdown.toggle additionalClassNames='wds-global-navigation__dropdown-toggle'}}
+		{{svg model.header.image class='wds-icon wds-icon-small'}}
+		<span class="wds-global-navigation__account-menu-caption">{{i18n model.header.title.key ns='design-system'}}</span>
+	{{/dropdown.toggle}}
+	{{#dropdown.content
+		additionalClassNames='wds-global-navigation__dropdown-content'
+		dropdownRightAligned=true
+	}}
+		<ul class="wds-list wds-has-lines-between">
```
Comment (kvas-damian) [inline front/main/app/templates/components/design-system/global-navigation-account-navigation.hbs]: I wonder if it'd be usefull to have `wds-list` component extracted.

File: front/main/app/templates/components/design-system/global-navigation-dropdown.hbs [hunk 3]
```diff
@@ -1,21 +1,28 @@
...
+		title=(if showAvatar model.header.username.value)
+	}}
+		{{#if showAvatar}}
+			<img class="wds-avatar wds-is-circle" src="{{model.header.url}}" alt="{{model.header.username.value}}" />
```
Comment (kvas-damian) [inline front/main/app/templates/components/design-system/global-navigation-dropdown.hbs]: Can we add `title` attribute here?

**Comment by kvas-damian** [comment]
Contextual components looks awesome!

File: front/main/app/templates/components/design-system/global-navigation-account-navigation.hbs (hunk 2 above)
Comment (rogatty) [inline front/main/app/templates/components/design-system/global-navigation-account-navigation.hbs]: There is no logic here so I see no major benefits in doing that. When in comes to downsides - keep in mind https://medium.com/square-corner-blog/optimizing-ember-templates-c479d26fe58e

File: front/main/app/templates/components/design-system/global-navigation-dropdown.hbs (hunk 3 above)
Comment (rogatty) [inline front/main/app/templates/components/design-system/global-navigation-dropdown.hbs]: It's already few lines above.

**Pull request merged**